File: backend/src/lib/pipeline-config.ts
import { graphIdFor, graphIdToTopic, userSlugFromNodeId } from './graph-topology';

/** Publisher and seeker are different end-users with separate graphs. */
export const PUBLISHER_USER_NODE_ID = process.env.COT_PUBLISHER_USER_ID ?? 'user_117';
export const SEEKER_USER_NODE_ID = process.env.COT_SEEKER_USER_ID ?? 'User_902';

export const PUBLISHER_USER_SLUG = userSlugFromNodeId(PUBLISHER_USER_NODE_ID);
export const SEEKER_USER_SLUG = userSlugFromNodeId(SEEKER_USER_NODE_ID);

export const PUBLISHER_GRAPH_ID =
  process.env.PUBLISHER_GRAPH_ID ?? graphIdFor(PUBLISHER_USER_SLUG, 'publisher');
export const SEEKER_GRAPH_ID =
  process.env.SEEKER_GRAPH_ID ?? graphIdFor(SEEKER_USER_SLUG, 'seeker');

export const PUBLISHER_AGENT_ID = `${PUBLISHER_USER_SLUG}.publisher`;
export const SEEKER_AGENT_ID = `${SEEKER_USER_SLUG}.seeker`;

/** @deprecated Per-user topics replaced by market.signals.public */
export const PUBLISHER_KAFKA_TOPIC =
  process.env.KAFKA_PUBLISHER_TOPIC ?? graphIdToTopic(PUBLISHER_GRAPH_ID);

/** @deprecated Use PUBLISHER_USER_NODE_ID */
export const DEFAULT_USER_NODE_ID = PUBLISHER_USER_NODE_ID;

File: backend/app/lib/pipeline_config.py
import os

from app.lib.graph_topology import graph_id_for, user_slug_from_node_id

MAIN_USER_NODE_ID = os.getenv("MAIN_USER_NODE_ID", "user_771")
MAIN_USER_SLUG = user_slug_from_node_id(MAIN_USER_NODE_ID)
MAIN_GRAPH_ID = os.getenv("MAIN_GRAPH_ID", graph_id_for(MAIN_USER_SLUG, "main"))
MAIN_AGENT_ID = f"{MAIN_USER_SLUG}.main"

PUBLISHER_USER_NODE_ID = os.getenv("COT_PUBLISHER_USER_ID", "user_117")
SEEKER_USER_NODE_ID = os.getenv("COT_SEEKER_USER_ID", "User_902")

PUBLISHER_USER_SLUG = user_slug_from_node_id(PUBLISHER_USER_NODE_ID)
SEEKER_USER_SLUG = user_slug_from_node_id(SEEKER_USER_NODE_ID)

PUBLISHER_GRAPH_ID = os.getenv(
    "PUBLISHER_GRAPH_ID", graph_id_for(PUBLISHER_USER_SLUG, "publisher")
)
SEEKER_GRAPH_ID = os.getenv("SEEKER_GRAPH_ID", graph_id_for(SEEKER_USER_SLUG, "seeker"))

PUBLISHER_AGENT_ID = f"{PUBLISHER_USER_SLUG}.publisher"
SEEKER_AGENT_ID = f"{SEEKER_USER_SLUG}.seeker"
